module/users/helper.js

```diff
@@ -467,6 +467,62 @@ module.exports = class UsersHelper {
           });
         }
 
+        let totalCount = 0;
+        let mergedData = [];
+        //fetch all targeted programs to users
+        let targetedPrograms = await programsHelper.forUserRoleAndLocation(
+          requestedData,
+          "", // not passing page size
+          "", // not passing page number
+          "",
+          ["_id"]
+        );
+        let targetedProgramIds = gen.utils.arrayOfObjectToArrayOfObjectId(targetedPrograms.data);
+
+        // check current program is targeted or not
+        if(!targetedProgramIds.includes(programId)){
+          let solutionIds = []
+          let importedSurveys = await surveyService.getImportedSurveys(token,programId)
+          importedSurveys = importedSurveys.result
+          let importedObservations = await surveyService.getImportedObservations(token,programId)
+          importedObservations = importedObservations.result
+          let importedProjects = await improvementProjectService.importedProjects(token,programId); 
+          
+          if(importedProjects.data.length> 0 && (type ==="" || type === constants.common.IMPROVEMENT_PROJECT)){
```

Comment:
@ankitpws if we know it is importedProjects why do we check type === constants.common.IMPROVEMENT_PROJECT